src/client-configuration.ts

```diff
@@ -80,33 +98,74 @@ export interface ClientConfiguration {
    */
   query_timeout_ms?: number;
 
+  /**
+   * A traceparent provided back via logging and telemetry.
+   * Must match format: https://www.w3.org/TR/trace-context/#traceparent-header
+   */
+  traceparent?: string;
+
+  /**
+   * Enable or disable typechecking of the query before evaluation. If no value
+   * is provided, the value of `typechecked` in the database configuration will
+   * be used.
+   */
+  typecheck?: boolean;
+}
+
+/**
+ * Options for queries. Each query can be made with different options.
+ */
+export interface QueryOptions {
+  /** Optional arguments. Variables in the query will be initialized to the
+   * value associated with an argument key.
+   */
+  arguments?: QueryValueObject;
```

Comment:
i don't think this makes sense in `QueryOptions`. This is directly related to text in a `query`. So it should be in `QueryRequest` and nowhere else.

Is this necessary to be here to support using arguments with `fql` formed queries?

If so does passing arguments give us something 

```
fql`${foo}`
``` 

does not?

Reply:
There are currently no ways for the user to add arguments, as in `QueryRequest.arguments`. We lost that when we changed the signature of `Client.query` from `(request: QueryRequest | Query, ...)` to (request: Query, ...)`

`QueryRequest` is now just a bit of wire protocol.

Reply:
OK let's leave it here then.

Reply:
Didn't mean to be so terse, but I got pulled away. 

I do think that it is an important use case to be able to provide arguments directly. For reference, one thing that comes to mind to me is GraphQL clients accepting "variables". That makes me wonder if we should call name the field `QueryOptions.variables` instead, but I think "arguments" is nice because it's basically a one-to-one what you pass in is sent as `QueryRequest.arguments` (just encoded for tagged format).

I hesitate to try and add it as a third parameter in order to keep `QueryOptions` separate from it.

I am willing to concede we can leave this out now and continue to discussion, I think it could be a clean addition in a later patch. You still okay with this API for arguments?

Reply:
i think its fine how you have it.